fili-core/src/main/java/com/yahoo/bard/webservice/data/dimension/impl/LuceneSearchProvider.java

```diff
@@ -77,6 +79,21 @@
             SYSTEM_CONFIG.getPackageVariableName("lucene_search_timeout_ms"), 600000
     );
 
+    public static final LRUQueryCache LUCENE_QUERY_CACHE;
+    static {
+        // Lucene 5.3.0 disables query caching.  Create cache for use here.
+        // min of 32MB or 5% of the heap size.  Same default as Lucene 5.4.0.
+        final long maxRamBytesUsed = SYSTEM_CONFIG.getLongProperty(
+            SYSTEM_CONFIG.getPackageVariableName("lucene_cache_size_bytes"),
+            Math.min(1L << 25, Runtime.getRuntime().maxMemory() / 20));
```

Comment:
I'd love to include an Eclipse checkstyle, but I'm not sure you'll be able to get it to match what IntelliJ has fully. I don't think Eclipse supports chop-wrapping of param lists. The only reason it's not included yet is that most of the active committers don't use Eclipse 😄 

So, feel free to submit one, and perhaps @will-lauer can take a look at it (since I think he uses Eclipse), and we'll get it added.

<sub>It's also been brought up that the existing IntelliJ config is a bit out of date vs. the Checkstyle config, so we need work on that too, unfortunately 😢 </sub>